### ISSUE TITLE
teething problems on initial setup

### ISSUE THREAD

**Issue by plastikfan** [mikepenz/release-changelog-builder-action#1039]
I am new to automated changelog generation wih GH actions, so please excuse my dumb questions

I have setup a [dummy](https://github.com/snivilised/lucretia) golang project, to use "[release-changelog-builder-action](https://github.com/mikepenz/release-changelog-builder-action)", so that I can make mistakes without fear of ruining a production ready repo.

I am using the default json config as per instructions, attempting to see what I can generate, but as you can see from my initial [run](https://github.com/snivilised/lucretia/actions/runs/4326064123), it failed.

Currently, I don't use pull requests to trigger builds, I have set it up to run on all tag pushes. The model for committing is simply to push a tag from a release branch (off master), but I think the default is expecting a PR. I don't know why its falling over in a heap, so I would be grateful if you could let me know better how to get this action to generate the change log based upon pushing a tag, rather than a pull request. (To be honest, I don't really understand the pull request model in terms of doing releases yet, perhaps I should change my process). To start off with, I just wan to get this to work, then I'll polish off/learn the PR process.

EDIT: Actually, I've taken another look and it looks like an authorisation problem. I changed the GITHUB_TOKEN to point to the secrets.GITHUB_TOKEN. I was using another personal access token before, but to simplify, changed to use the GH actions token instead. But this results in the same error.

Thanks.

**Comment by plastikfan** [comment]
I found out what was failing, I needed to add write access:

```
    permissions:
      contents: write
      pull-requests: write
```

I need to read up on the whole release process now, if you have any tips on resources about setting up a release sequence, I'd be interested to know. The github documentation is just reference information, where as what I would be looking for is holistic information that describes the whole process so I can create releases smoothly and automated as much as possible.

Great repo by the way.
